src/simulation/LoadGenerator.cpp

```diff
@@ -258,21 +262,69 @@ LoadGenerator::generateLoad(GeneratedLoadConfig cfg)
 
     for (int64_t i = 0; i < txPerStep; ++i)
     {
-        switch (cfg.mode)
+        if (cfg.mode == LoadGenMode::CREATE)
         {
-        case LoadGenMode::CREATE:
             cfg.nAccounts = submitCreationTx(cfg.nAccounts, cfg.offset,
                                              cfg.batchSize, ledgerNum);
+        }
+        else
+        {
+            auto sourceAccountId =
+                rand_uniform<uint64_t>(0, cfg.nAccounts - 1) + cfg.offset;
+
+            std::function<
+                std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>()>
+                generateTx;
+
+            switch (cfg.mode)
+            {
+            case LoadGenMode::CREATE:
+                releaseAssert(false);
+                break;
+            case LoadGenMode::PAY:
+                generateTx = [&]() {
+                    return paymentTransaction(cfg.nAccounts, cfg.offset,
```

Comment:
I think asan complains that you're capturing `sourceAccountId` and `opCount` but those variables go out of scope

Reply:
Nice, your change has already saved me a couple of hours debugging time!